js/windowMgr.js

```diff
@@ -357,14 +347,28 @@ function doCreateMainWindow(initialUrl, initialBounds) {
 
                         electron.dialog.showMessageBox(options, function (index) {
                             if (index === 0) {
-                                mainWindow.reload();
+                                browserWin.reload();
                             }
                             else {
-                                mainWindow.close();
+                                browserWin.close();
                             }
                         });
                     });
 
+                    // In case we navigate to an external link from inside a pop-out,
+                    // we open that link in an external browser rather than creating
+                    // a new window
+                    browserWin.webContents.on('new-window', (childEvent, childWinUrl) => {
+                        childEvent.preventDefault();
+                        openUrlInDefaultBrowser(childWinUrl);
+                    });
+
+                    // Clear up the browser window once the window is closed
+                    // to avoid leakage
+                    browserWin.on('closed', () => {
```

Comment:
there is already a close above on line 334.  there you can remove event handler.
how will setting browserWin to null clean up?